File: java-code/src/test/java/com/vd/study/java_code/observer/TestObservable.java
package com.vd.study.java_code.observer;

public class TestObservable extends Observable<String> {

    public TestObservable(String value) {
        super(value);
    }

    int observersCount = 0;

    @Override
    public void register(Observer<String> observer) {
        super.register(observer);
        observersCount++;
    }

    @Override
    public void unregister(Observer<String> observer) {
        super.unregister(observer);
        observersCount--;
    }
}

File: kotlin-code/src/test/java/com/vd/study/kotlin_code/observer/TestObservable.kt
package com.vd.study.kotlin_code.observer

class TestObservable(value: String) : Observable<String>(value) {

    var observersCount: Int = 0

    override fun register(observer: Observer<String>) {
        super.register(observer)
        observersCount++
    }

    override fun unregister(observer: Observer<String>) {
        super.unregister(observer)
        observersCount--
    }
}